packages/Meshfree/src/DTK_SplineOperator_def.hpp

```diff
@@ -0,0 +1,300 @@
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef DTK_SPLINE_OPERATOR_DEF_HPP
+#define DTK_SPLINE_OPERATOR_DEF_HPP
+
+#include <ArborX.hpp>
+#include <BelosBlockGmresSolMgr.hpp>
+#include <BelosLinearProblem.hpp>
+#include <BelosTpetraAdapter.hpp>
+#include <DTK_DBC.hpp>
+#include <DTK_DetailsMovingLeastSquaresOperatorImpl.hpp>
+#include <DTK_DetailsNearestNeighborOperatorImpl.hpp> // fetch
+#include <DTK_DetailsPolynomialMatrix.hpp>
+#include <DTK_DetailsSplineProlongationOperator.hpp>
+
+#include <Stratimikos_DefaultLinearSolverBuilder.hpp>
+#include <Teuchos_XMLParameterListCoreHelpers.hpp>
+#include <Thyra_DefaultAddedLinearOp.hpp>
+#include <Thyra_DefaultMultipliedLinearOp.hpp>
+#include <Thyra_DefaultScaledAdjointLinearOp.hpp>
+#include <Thyra_LinearOpWithSolveFactoryHelpers.hpp>
+#include <Thyra_TpetraThyraWrappers.hpp>
+
+namespace DataTransferKit
+{
+
+template <typename DeviceType, typename CompactlySupportedRadialBasisFunction,
+          typename PolynomialBasis>
+Teuchos::RCP<
+    typename SplineOperator<DeviceType, CompactlySupportedRadialBasisFunction,
+                            PolynomialBasis>::Operator>
+SplineOperator<DeviceType, CompactlySupportedRadialBasisFunction,
+               PolynomialBasis>::
+    buildBasisOperator(
+        Teuchos::RCP<const Map> domain_map, Teuchos::RCP<const Map> range_map,
+        Kokkos::View<Coordinate const **, DeviceType> source_points,
+        Kokkos::View<Coordinate const **, DeviceType> target_points,
+        int const knn )
+{
+    auto teuchos_comm = domain_map->getComm();
+    auto teuchos_mpi_comm =
+        Teuchos::rcp_dynamic_cast<const Teuchos::MpiComm<int>>( teuchos_comm );
+    MPI_Comm comm = ( *teuchos_mpi_comm->getRawMpiComm() )();
+
+    int const num_source_points = source_points.extent( 0 );
+    int const num_points = target_points.extent( 0 );
+
+    ArborX::DistributedSearchTree<DeviceType> distributed_tree( comm,
+                                                                source_points );
+    DTK_CHECK( !distributed_tree.empty() );
+
+    // Perform the actual search.
+    auto queries =
+        Details::MovingLeastSquaresOperatorImpl<DeviceType>::makeKNNQueries(
+            target_points, knn );
+
+    Kokkos::View<int *, DeviceType> offset( "offset", 0 );
+    Kokkos::View<int *, DeviceType> ranks( "ranks", 0 );
+    Kokkos::View<int *, DeviceType> indices( "indices", 0 );
+    distributed_tree.query( queries, indices, offset, ranks );
```

Comment:
Fine but remember we are deprecating that overload.

Reply:
Out of scope.

Reply:
I am not objecting to us merging the way it is but I will mark it as unresolved.  We chose to ignore.